src/Icicle/Source/Checker/Invariants.hs

```diff
@@ -122,11 +122,11 @@ invariantX ctx x
      -> invariantX ctx s >> mapM_ (invariantX ctx . snd) ps
 
  where
-  goFun n args
+  goFun a n args
    | Just fun <- Map.lookup n $ checkBodies ctx
    = let ctx' = foldl bindArg ctx (arguments fun `zip` args)
-     -- TODO: wrap error in function information
-     in  invariantQ ctx' $ body fun
+     in  errorInFunctionEither a n
```

Comment:
The idea is right, but the rest is wrong: it turns out that "checkBodies" is always empty, and the only reason I/we *thought* this was working, was because invariant check is also run after inlining.